parastell/magnet_coils.py

```diff
@@ -401,164 +386,110 @@ class MagnetCoil(object):
-
-        # Determine part of rotated vector parallel to original
-        rot_par = np.cos(rot_ang_norm)
-        # Determine part of rotated vector orthogonal to original
-        rot_perp = np.sin(rot_ang_norm)
+    def __init__(self, coords, tangents, center_of_mass, width, thickness):
 
-        # Compute orthogonal part of thickness vector after rotation
-        t_vec_perp_rot = rot_par * t_vec_perp + rot_perp * orth
-        # Compute thickness vector after rotation
-        t_vec_rot = normalize(t_vec_perp_rot + t_vec_par)
-
-        # Orient cross-section in its plane such that it faces the global origin
-
-        # Extract initial path point
-        pos = cubit.vertex(path_origin).coordinates()
-
-        # Project position vector onto cross-section
-        pos_proj = normalize(pos - np.inner(pos, norm) * norm)
-
-        # Compute angle by which to rotate cross-section such that it faces the
-        # origin
-        rot_ang_orig = np.arccos(np.inner(pos_proj, t_vec_rot))
-
-        # Re-orient rotated cross-section such that thickness vector faces
-        # origin
-        cubit.cmd(
-            f"rotate Surface {surf_id} angle {np.rad2deg(rot_ang_orig)} about "
-            "origin 0 0 0 direction " + " ".join(str(i) for i in norm)
-        )
+        self.coords = coords
+        self.tangents = tangents
+        self.center_of_mass = center_of_mass
+        self.width = width
+        self.thickness = thickness
 
     def create_magnet(self):
-        """Creates magnet coil volumes in cubit.
+        """Creates a single magnet coil CAD solid in CadQuery.
 
         Returns:
-            volume_id (int): magnet volume ids in cubit
+            coil (object): cq.Solid object representing a single magnet coil.
         """
-        # Cross-section inititally populated with thickness vector
-        # oriented along x axis
-        t_vec = np.array([1, 0, 0])
+        tangent_vectors = [
+            cq.Vector(tuple(tangent)) for tangent in self.tangents
+        ]
 
-        # Create cross-section for sweep
-        cubit.cmd(f"create surface " + self.shape_str + " zplane")
+        # Define coil filament path normals such that they face the filament
+        # center of mass
+        normal_dirs = np.array([i - self.center_of_mass for i in self.coords])
+        normal_dirs = (
+            normal_dirs / np.linalg.norm(normal_dirs, axis=1)[:, np.newaxis]
+        )
 
-        # Store cross-section index
-        cs_id = cubit.get_last_id("surface")
-        # Cross-section initially populated with normal oriented along z
-        # axis
-        cs_axis = np.array([0, 0, 1])
+        # Project normal directions onto desired coil cross-section (CS) plane
+        # at each filament position to define true filament path normals
+        parallel_parts = []
+        for dir, tangent in zip(normal_dirs, self.tangents):
+            parallel_parts.append(np.dot(dir, tangent))
+        parallel_parts = np.array(parallel_parts)
```

Comment:
If these are numpy arrays then this block becomes
```suggestion
        parallel_parts = np.dot(normal_dirs, self.tangents)
```
Or close (maybe need to specify an axis?)

Reply:
As mentioned above, these are all numpy arrays. Unfortunately, the numpy functions that perform the dot product, `dot` and `inner`, do not have a way of specifying an axis along which to operate. Without a specified axis, they will just perform a full matrix operation.

Reply:
Can we structure the matrices such that a full matrix operation is the right thing to do?